File: examples/java/src/main/java/examples/svg/SVGGraphicsFactory.java
package examples.svg;

import org.jetbrains.annotations.NotNull;
import qrcode.render.QRCodeGraphics;
import qrcode.render.QRCodeGraphicsFactory;

public class SVGGraphicsFactory extends QRCodeGraphicsFactory {
    /**
     * IntelliJ might show an error on this method, but it can be ignored. Not sure what causes it.
     */
    @NotNull
    @Override
    public QRCodeGraphics newGraphics(int width, int height) {
        return new SVGQRCodeGraphics(width, height);
    }
}

File: examples/kotlin/src/main/kotlin/svg/SVGGraphicsFactory.kt
package svg

import qrcode.render.QRCodeGraphics
import qrcode.render.QRCodeGraphicsFactory

/**
 * Class that'll create the [SVGQRCodeGraphics] instances that'll be used to draw the QRCode
 */
class SVGGraphicsFactory : QRCodeGraphicsFactory() {
    override fun newGraphics(width: Int, height: Int): QRCodeGraphics =
        SVGQRCodeGraphics(width, height)
}
